From 4850a91097bafe71a38333d9532aa22a0698baf0 Mon Sep 17 00:00:00 2001
From: Nate Jiang <qiangjiang@google.com>
Date: Thu, 29 May 2025 18:32:53 +0000
Subject: [PATCH] Do not treat SdkSandBox as privileges App

Flag: EXEMPT bug fix
Bug: 399885815
Test: atest com.android.server.wifi
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:b8feb0cc861f6071e3d3e5e289ce969237901722)
Merged-In: I1ee7a09b7628a67dcb3502b29d99761bafd913b5
Change-Id: I1ee7a09b7628a67dcb3502b29d99761bafd913b5
---
 .../java/com/android/server/wifi/util/WifiPermissionsUtil.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/service/java/com/android/server/wifi/util/WifiPermissionsUtil.java b/service/java/com/android/server/wifi/util/WifiPermissionsUtil.java
index 7c79fafa9d..a053ec15bb 100644
--- a/service/java/com/android/server/wifi/util/WifiPermissionsUtil.java
+++ b/service/java/com/android/server/wifi/util/WifiPermissionsUtil.java
@@ -1138,6 +1138,9 @@ public class WifiPermissionsUtil {
     public boolean isSystem(String packageName, int uid) {
         long ident = Binder.clearCallingIdentity();
         try {
+            if (SdkLevel.isAtLeastT() && Process.isSdkSandboxUid(uid)) {
+                return false;
+            }
             ApplicationInfo info = mContext.getPackageManager().getApplicationInfoAsUser(
                     packageName, 0, UserHandle.getUserHandleForUid(uid));
             return (info.flags & APP_INFO_FLAGS_SYSTEM_APP) != 0;
-- 
2.50.1.351.gcf940e82a1-goog

